libgalois/src/PropertyGraph.cpp

```diff
@@ -67,17 +67,37 @@ MapTopology(const tsuba::FileView& file_view) {
 
   auto* out_dests = reinterpret_cast<uint32_t*>(out_indices + num_nodes);
 
-  auto indices_buffer = std::make_shared<arrow::MutableBuffer>(
-      reinterpret_cast<uint8_t*>(out_indices), num_nodes);
+  auto adj_arr = std::make_unique<katana::LargeArray<uint64_t> >();
+  auto dst_arr = std::make_unique<katana::LargeArray<uint32_t> >();
 
-  auto dests_buffer = std::make_shared<arrow::MutableBuffer>(
-      reinterpret_cast<uint8_t*>(out_dests), num_edges);
+  // adj_arr->allocateBlocked(num_nodes);
+  // dst_arr->allocateBlocked(num_edges);
+
+  adj_arr->allocateInterleaved(num_nodes);
+  dst_arr->allocateInterleaved(num_edges);
+
+  katana::ParallelSTL::copy(out_indices, out_indices + num_nodes, adj_arr->begin());
+  katana::ParallelSTL::copy(out_dests, out_dests + num_edges, dst_arr->begin());
+
+  // Some sanity checks on topology
+  katana::do_all (katana::iterate(*dst_arr), [&] (auto n) {
+    (void)n; // to suppress unused var warning in release builds
```

Comment:
nit: I prefer `([[maybe_unused]] auto n)` as an argument because that's more common in the code.

Reply:
done.